### PR TITLE
please add hostingde

### PR DESCRIPTION
**Pull request by initit**
Please add hostingde.

Thanks
Michael

### CODE REVIEW

**Comment by CapsuleCD** [comment]
Hi.

I'm an automated pull request bot named [CapsuleCD](https://www.capsulecd.com). I handle testing, versioning and package releases for this project. 
- If you're the owner of this repo, you can click the button below to kick off the CapsuleCD build pipeline and create an automated release.'
- If not, don't worry, someone will be by shortly to check on your pull request. 

[CapsuleCD](https://www.capsulecd.com/project/github/AnalogJ/lexicon/pullrequests/427)

---
If you're interested in learning more about [CapsuleCD](http://www.github.com/AnalogJ/capsulecd), or adding it to your project, you can check it out [here](https://www.capsulecd.com)

**Comment by initit** [comment]
Is there anything i do to speed up or support this PR?

Thanks & regards 
Michael

File: lexicon/providers/hostingde.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+        processed_records = []
+
+        for record in raw_records:
+            processed_record = {
+                'type': record['type'],
+                'name': self._full_name(record['name']),
+                'ttl': record['ttl'],
+                'id': record['id'],
+                'content': record['content'][1:-1] if record['type'] == 'TXT'
```
Comment (adferrand) [inline lexicon/providers/hostingde.py]: You can reuse the `_clean_TXT_record` method defined by superclass `BaseProvider` here.

File: tests/providers/test_hostingde.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    def _filter_post_data_parameters(self):
+        return ['authToken']
+
+    def _filter_headers(self):
+        return ['Authorization']
+
+    def _filter_query_parameters(self):
+        return ['secret_key']
+
+    def _filter_response(self, response):
```
Comment (adferrand) [inline tests/providers/test_hostingde.py]: It is not necessary to override this method. Default behavior in superclass `IntegrationTests` is already to return `response` untouched.

**Pull request merged**